File: crisp/core/src/test/java/org/onehippo/cms7/crisp/core/resource/jdom/model/Image.java
/*
 *  Copyright 2017 Hippo B.V. (http://www.onehippo.com)
 * 
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 * 
 *       http://www.apache.org/licenses/LICENSE-2.0
 * 
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package org.onehippo.cms7.crisp.core.resource.jdom.model;

import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement(name="image")
public class Image {

    private String source;
    private String name;
    private int hOffset;
    private int vOffset;
    private String alignment;

    @XmlAttribute(name="src")
    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    @XmlAttribute(name="name")
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @XmlElement(name="hOffset")
    public int gethOffset() {
        return hOffset;
    }

    public void sethOffset(int hOffset) {
        this.hOffset = hOffset;
    }

    @XmlElement(name="vOffset")
    public int getvOffset() {
        return vOffset;
    }

    public void setvOffset(int vOffset) {
        this.vOffset = vOffset;
    }

    @XmlElement(name="alignment")
    public String getAlignment() {
        return alignment;
    }

    public void setAlignment(String alignment) {
        this.alignment = alignment;
    }

}

File: spa-sdk/packages/spa-sdk/src/page/image.ts
/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { injectable, inject } from 'inversify';
import { Builder } from './factory';
import { LinkFactory } from './link-factory';
import { Link } from './link';

export const ImageFactory = Symbol.for('ImageFactory');
export const ImageModelToken = Symbol.for('ImageModelToken');

export type ImageFactory = Builder<[ImageModel], Image>;

/**
 * @hidden
 */
type ImageLinks = 'site';

/**
 * Image model.
 * @hidden
 */
export interface ImageModel {
  displayName: string;
  fileName: string | null;
  height: number;
  links: Partial<Record<ImageLinks, Link>>;
  mimeType: string;
  name: string;
  size: number;
  width: number;
}

export interface Image {
  /**
   * @return The image display name.
   */
  getDisplayName(): string;

  /**
   * @return The image file name.
   */
  getFileName(): string | undefined;

  /**
   * @return The image height.
   */
  getHeight(): number;

  /**
   * @return The image mime-type.
   */
  getMimeType(): string;

  /**
   * @return The image name.
   */
  getName(): string;

  /**
   * @return The image size.
   */
  getSize(): number;

  /**
   * @return The image link.
   */
  getUrl(): string | undefined;

  /**
   * @return The image width.
   */
  getWidth(): number;
}

@injectable()
export class ImageImpl implements Image {
  constructor(
    @inject(ImageModelToken) protected model: ImageModel,
    @inject(LinkFactory) private linkFactory: LinkFactory,
  ) {}

  getDisplayName(): string {
    return this.model.displayName;
  }

  getFileName(): string | undefined {
    return this.model.fileName ?? undefined;
  }

  getHeight(): number {
    return this.model.height;
  }

  getMimeType(): string {
    return this.model.mimeType;
  }

  getName(): string {
    return this.model.name;
  }

  getSize(): number {
    return this.model.size;
  }

  getUrl(): string | undefined {
    return this.model.links.site && this.linkFactory.create(this.model.links.site);
  }

  getWidth(): number {
    return this.model.width;
  }
}
